### PR TITLE
Cart Functionalities

### PR DESCRIPTION
**Pull request by GabrielDushime**
What does this PR do?
------------------------
This PR enhances the functionality related to the shopping cart feature. It includes additions, updates, and improvements to the cart management system, covering operations such as adding items to the cart, viewing the cart contents, updating cart items, and clearing the cart.

Describe your changes
-------------------------
-Addition of Cart Management Services: Implemented services for adding, viewing, updating, and clearing the shopping cart, providing robust functionality for managing cart items.

-Integration with Database Models: Integrated the cart services with the database models to ensure seamless interaction between the application logic and the database.

-Endpoint Implementations: Created endpoint functions to handle HTTP requests for adding items to the cart, viewing cart contents, updating cart items, and clearing the cart.

-Test Coverage: Added comprehensive test cases to ensure the correctness and reliability of the cart management functionality, covering various scenarios and edge cases.

-Documentation Updates: Included documentation updates to reflect the changes made to the cart-related functionality, ensuring that developers have clear guidance on using and understanding the new features.

PR Type
What kind of change does this PR introduce?
-------------------------------------------------
[x] Feature
[ ] Bugfix
[ ] Refactoring (no functional changes, no API changes)
[ ] Documentation content changes
[ ] Tests
[ ] Other

How should this be manually tested?
----------------------------------------
-Add items to the cart using the provided endpoint and verify that the items are correctly added.

-View the cart contents through the appropriate endpoint and ensure that all added items are displayed accurately.

-Update the quantity of items in the cart and verify that the changes reflect correctly in the cart.

-Clear the cart using the designated endpoint and confirm that the cart is emptied as expected.

### CODE REVIEW

**Review by Mugisha146 — CHANGES_REQUESTED** [review]
1.ensure userId is extracted from the authenticated user session, not directly from req.body  2.consider using async/await for CartService.createCart to improve readability                              3.properly handle the case where the cart is not found

File: src/controllers/cart.controller.ts [hunk 1]
```diff
@@ -0,0 +1,163 @@
+import { Request, Response } from "express";
+import { CreateCartAttributes, CreateCartItemAttributes } from "../types/cart.types";
+/* import Cart from "../database/models/cart.model";  */
+import { CartService } from "../services/cart.services";
+import { ProductService } from "../services/Product.services";
+
+
+export const createCart = async (req: Request, res: Response) => {
+    try {
+      const { name, description, userId } = req.body;
```
Comment (mugishaj092) [inline src/controllers/cart.controller.ts:10]: userId must come from req.user instead req.body

**Review by mugishaj092 — CHANGES_REQUESTED** [review]

File: src/controllers/cart.controller.ts (hunk 1 above)
Comment (GabrielDushime) [reply src/controllers/cart.controller.ts]: I have resolved it you can check

**Comment by GabrielDushime** [comment]
> 1.ensure userId is extracted from the authenticated user session, not directly from req.body 2.consider using async/await for CartService.createCart to improve readability 3.properly handle the case where the cart is not found

I have resolved it you can check

File: src/database/migrations/20240614113048-create-cart.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+      },
+      description: {
+        type: Sequelize.STRING,
+        allowNull: false,
+      },
+      userId: {
+        type: Sequelize.UUID,
+        allowNull: false,
+        references: {
+          model: 'Users',
```
Comment (mugishaj092) [inline src/database/migrations/20240614113048-create-cart.js:24]: here you was suppose to use users instead Users

File: src/database/migrations/create-carts.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
```
Comment (mugishaj092) [inline src/database/migrations/create-carts.js:3]: This file is not necessary

File: src/database/models/cart.model.ts [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+Cart.init(
+  {
+    id: {
+      type: DataTypes.UUID,
+      allowNull: false,
+      primaryKey: true,
+      defaultValue: DataTypes.UUIDV4,
+    },
+    name: {
```
Comment (mugishaj092) [inline src/database/models/cart.model.ts:34]: I don't get the role of name and description

**Review by mugishaj092 — CHANGES_REQUESTED** [review]

**Review by mugishaj092 — APPROVED** [review]
You did great job bruh

**Review by jamesndekezi — APPROVED** [review]
Well done

**Review by SHYAKA-Aime — APPROVED** [review]
Working well, great work

**Review by vegetason — APPROVED** [review]
Good work, the required functionalities work as expected
